catalog/opensearch/catalog-opensearch-source/src/main/java/org/codice/ddf/opensearch/source/OpenSearchSource.java

```diff
@@ -957,4 +968,78 @@ private WebClient newOpenSearchClient(String url, Subject subj) {
     }
     return metacards;
   }
+
+  protected class SpatialSearch {
+    private final PointRadiusSearch pointRadiusSearch;
+    private final Polygon polygonSearch;
+
+    public SpatialSearch(PointRadiusSearch pointRadiusSearch) {
+      notNull(
+          pointRadiusSearch,
+          "Cannot construct a spatial search with just a null point-radius search");
+      this.pointRadiusSearch = pointRadiusSearch;
+      polygonSearch = null;
+    }
+
+    public SpatialSearch(Polygon polygonSearch) {
+      notNull(polygonSearch, "Cannot construct a spatial search with just a null polygon search");
+      pointRadiusSearch = null;
+      this.polygonSearch = polygonSearch;
+    }
+
+    @Nullable
+    public PointRadiusSearch getPointRadiusSearch() {
+      return pointRadiusSearch;
+    }
+
+    @Nullable
+    public Polygon getPolygonSearch() {
+      return polygonSearch;
+    }
+  }
+
+  /**
+   * Method to combine spatial searches into either a polygon or a point-radius search. OpenSearch
+   * endpoints will ignore multiple spatial query parameters. This method is protected so that
+   * downstream projects may try to implement another algorithm (e.g. best-effort) to combine
+   * searches.
+   *
+   * @return null if the searches cannot be combined, or a {@link SpatialSearch} with one {@link
+   *     PointRadiusSearch} or one {@link Polygon} that is the combination of all of the searches
+   */
+  protected Optional<SpatialSearch> createCombinedSpatialSearch(
+      final Queue<PointRadiusSearch> pointRadiusSearches, final Queue<Polygon> polygonSearches) {
+    final boolean filterContainedSomePointRadiusSearchCriteria =
+        CollectionUtils.isNotEmpty(pointRadiusSearches);
+    final boolean filterContainedSomePolygonSearchCriteria =
+        CollectionUtils.isNotEmpty(polygonSearches);
+
+    if (filterContainedSomePointRadiusSearchCriteria && filterContainedSomePolygonSearchCriteria) {
+      LOGGER.debug(
```

Comment:
Hope this doesn't surprise anyone since we are dropping all spatial terms and not taking the last one

Reply:
Right. Before this PR, the last point-radius filter in the query would be used to populate the query parameters. If there were no point-radius filters, then the  last polygon filter in the query would be used to populate the query parameters.
https://github.com/codice/ddf/blob/b279c781e127c5fc9eb73956b9b79609f6b5a5cd/catalog/opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/impl/OpenSearchSource.java#L979-L996
The OpenSearch Source was dropping spatial filters without any indication and populating the query parameters from the filters in an unexpected way. This logic was updated to not support any filters with multiple spatial filters. This implementation could be updated in the future or in downstream projects to, for example, combine all spatial criteria into a single `Geometry`.